src/containers/AuthenticationPage/AuthenticationPage.js

```diff
@@ -194,8 +201,12 @@ export const AuthenticationForms = props => {
 
   // eslint-disable-next-line no-confusing-arrow
   const errorMessage = (error, message) => (error ? message : null);
+
+  const loginOrIdpAuthErrorMessage = (loginError, idpAuthError) =>
+    errorMessage(idpAuthError, idpAuthErrorMessage) || errorMessage(loginError, loginErrorMessage);
+
   const loginOrSignupError = isLogin
-    ? errorMessage(loginError, loginErrorMessage)
+    ? loginOrIdpAuthErrorMessage(loginError, idpAuthError)
     : errorMessage(signupError, signupErrorMessage);
```

Comment:
How about using pure optional chaining instead?

```js
  const hasError = error => !!error;
  const loginOrSignupError = isLogin && hasError(loginError)
    ? loginErrorMessage
    : isLogin && hasError(idpAuthError)
    ? idpAuthErrorMessage
    : hasError(signupError)
    ? signupErrorMessage
    : null;
```

Reply:
Maybe with the exception that leave out the `hasError` fn and just use `!!` in the chaining.